src/mem/mem_ctrl.hh

```diff
@@ -106,8 +105,12 @@ class MemPacket
 
     const bool read;
 
+    //every write will initiate a read first
+    bool read_before_write;
+
     /** Does this packet access DRAM?*/
-    const bool dram;
+    //const bool dram;
```

Comment:
Why no longer const?

I think probably because you're re-using the same packet throughout the lifetime of the request. I am not convinced this is the best way to do it. I would have expected you to create a new DRAMPacket for each "step"